File: Backend/Lulus.Data/Entities/Product.cs
using Lulus.Data.Enums;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Lulus.Data.Entities
{
    public class Product: BaseAttribute
    {
        public string Name { get; set; }
        public double Price { get; set; }
        public string Description { get; set; }
        public int CategoryID { get; set; }
        public int DesignerID { get; set; }

        public ProductStatus Status { get; set; }
        public LengthState? LengthState { get; set; }
        public SleeveLengthState? SleeveLengthState { get; set; }
        public virtual List<Feedback> Feedbacks { get; set; }

        public virtual Category Category { get; set; }
        public virtual List<ProductLine> ProductLines { get; set; }
        public virtual List<Product_SubCategory> Product_SubCategories { get; set; }
        public virtual List<Product_Occation> Product_Occations { get; set; }
        public virtual Desinger Desinger { get; set; }
        public virtual List<FavoriteProduct> FavoriteProducts { get; set; }
        public virtual List<Sale> Sales { get; set; }
    }
}

File: Frontend/revolve/src/app/model/Product.ts
import { Feedback } from './Feedback';
import {ProductLine} from './ProductLine';
export class Product{
    public id: number;
    public name: String;
    public price: number;
    public description:String;
    public categoryID: number;
    public designerID: number;
    public status: number;
    public averageStarLeft: number[];
    public averageStarRight: number[];
    public productLines: ProductLine[];
    public feedbacks: Feedback[];

    constructor(id:number, name: String, price: number, description: String, cateID: number, designerID: number,status: number, averageStar: number){
        this.id = id;
        this.name = name;
        this.price = price;
        this.description = description;
        this.categoryID = cateID;
        this.designerID = designerID;
        this.status = status;
        this.averageStarLeft = Array(averageStar).fill(4);
        this.averageStarRight = Array(5 - averageStar).fill(4)
        this.productLines = new Array<ProductLine>();
        this.feedbacks = new Array<Feedback>();
    }
}